src/libstd/fs.rs

```diff
@@ -1724,6 +1734,23 @@ mod tests {
         assert!(canary.exists());
     }
 
+    // FIXME(#12795) depends on lstat to work on windows
+    #[cfg(not(windows))]
```

Comment:
Oh, nevermind, the test won't work because soft_link doesn't do the right thing on Windows.

Actually, given that this is a new test, please avoid using `soft_link` because it's deprecated.